LanguageFeatures/Augmentation-libraries/augmenting_constructors_A11_t03.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - Initializer lists are not re-run, they have already executed and
+///     shouldn't be executed twice. The same goes for initializing formals and
+///     super parameters.
+///
+/// @description Checks that when `augmented()` is called in the body of an
+/// augmenting constructor super parameters are not re-run.
```

Comment:
The spec wording isn't too good here. I don't think we can say that a super parameter "runs". It's basically a syntactic sugar mechanism, and the meaning of a super-parameter declaration is a regular parameter plus a type (which may have been written or may come from the superclass) plus possibly a default value (written or from the superclass), and an actual argument in the superinitializer, `super.maybeAName(some, arguments)`.

I think we have to understand "the same goes for ... super parameters" as "a positional super-parameter will not be passed twice in the implicitly modified or induced superinitializer".

Reply:
This could again be tested by detecting that we don't get anything like "wrong number of positional parameters in superinitializer" when `super.x` occurs both in the introductory and an augmenting declaration, but also that it _is_ an error if the number of positional parameters to the superinitializer will only be _correct_ in the case where the super-parameter is passed twice (that is, we would then expect a compile-time error because there are too few positional parameters).

For named parameters we can only test the non-error case: If the same named parameter is passed twice then it's an error, so we just check that there is no error when the super-parameter is specified both in the introductory and an augmenting declaration.

Reply:
Same